src/main/java/use_case/execute_buy/ExecuteBuyDataAccessInterface.java

```diff
@@ -3,9 +3,9 @@
 import entity.User;
 
 public interface ExecuteBuyDataAccessInterface {
-    User getUserWithCredential(String credential) throws ValidationException;
+    User getUserWithCredential(String credential) throws ValidationError;
 
     // TODO: define this exception elsewhere
-    class ValidationException extends Exception {
+    class ValidationError extends Exception {
```

Comment:
Change this exception name back to `ValidationException`. In `Java`, `Error` usually implies a subclass of `Error`, a type used to indicate a serious system-level error that cannot be recovered (such as out of memory). `Exception` indicates subclass of `Exception`, an exception intended to be handled in the program.